llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -3330,16 +3330,17 @@ LegalizerHelper::LegalizeResult LegalizerHelper::lowerLoad(GAnyLoad &LoadMI) {
   LLT MemTy = MMO.getMemoryType();
   MachineFunction &MF = MIRBuilder.getMF();
 
-  unsigned MemSizeInBits = MemTy.getSizeInBits();
-  unsigned MemStoreSizeInBits = 8 * MemTy.getSizeInBytes();
+  unsigned MinMemSizeInBits = MemTy.getSizeInBits().getKnownMinValue();
```

Comment:
I think we can do this as `TypeSize MemSizeInBits` and `TypeSize MinMemStoreSizeInBits` as Craig Pointed out, which allows us to avoid calling getKnownMinValue. That way we avoid the case comparing fixed vector and scalable vector.

Reply:
Understood.
I used `getKnownMinValue` because we have cases for doing scalar arithmetic operations below. For example:
https://github.com/llvm/llvm-project/blob/9ce3a848fddb3cdc40f47ae35dae205d49459fb3/llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp#L3397-L3411

 `bit_floor` has to take a scalar.

So are you suggesting we make them `TypeSize MemSizeInBits` but call `getKnownMinValue` on demand?